### PR TITLE
Bump CI build from JDK 19 to 21 (LTS)

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
Secretly hoping it fixes https://github.com/square/okhttp/actions/runs/7313047455/job/19924201870

and we get off to the new year with clean builds.

File: okhttp/src/test/java/okhttp3/HttpUrlTest.kt [hunk 1]
```diff
@@ -560,7 +561,8 @@ open class HttpUrlTest {
         ']'.code
       )
       .override(
-        Encoding.IDENTITY,
+        // java.net.URL got stricter
+        if (PlatformVersion.majorVersion >= 21)  Encoding.SKIP else Encoding.IDENTITY,
```
Comment (yschimke) [inline okhttp/src/test/java/okhttp3/HttpUrlTest.kt:565]: @swankjesse I'm not sure this is right, is there a better way to make assertions here? That it now fails for URL?

**Comment by yschimke** [comment]
@swankjesse requesting post review.  Particularly over how to handle JDK 21 URI strictness.

**Comment by yschimke** [comment]
Landing to get builds green.

File: okhttp/src/test/java/okhttp3/HttpUrlTest.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/test/java/okhttp3/HttpUrlTest.kt:565]: Lemme attempt a follow-up. I think we need to fix our code.

**Review by swankjesse — COMMENTED** [review]
LGTM

File: okhttp/src/test/java/okhttp3/HttpUrlTest.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/test/java/okhttp3/HttpUrlTest.kt:565]: https://github.com/square/okhttp/pull/8363

**Pull request merged**